niflow_manager/cli/build.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+import subprocess as sp
+import tempfile, os, json
+from pathlib import Path
+from copy import deepcopy
+import yaml
+import neurodocker as ndr
+
+# default setting for specific neurodocker keys,
+# this value will not have to be set in the parameters.yml, but can be overwritten
+DEFAULT_INSTRUCTIONS = {
+    "miniconda": {'create_env': 'testkraken', 'activate': True}
+}
+# all keys allowed by neurodocker for Docker
+VALID_DOCKER_KEYS = ndr.Dockerfile._implementations.keys()
+
+
+def neurodocker_dict(workflow_path):
+    """reading spec.yml and creating neurodocker dict
+        for the environment defined in the requirements and post_build parts
+    """
+    with (workflow_path / 'spec.yml').open() as f:
+        params = yaml.safe_load(f)
+    instructions = []
+
+    # treating requirement as one env spec
+    params_env = params["requirements"]
+    if "base" not in params_env.keys():
+        raise ValueError("base image has to be provided")
+
+    miniconda_env = None
+    for key, spec in params_env.items():
+        if key == "base":
+            base_image = spec.get('image', None)
+            if base_image is None:
+                raise Exception("image has to be provided in base")
+            this_instruction = ('base', base_image)
+            pkg_manager = spec.get('pkg_manager', None)
+            if pkg_manager is None:
+                if base_image in ["centos", "fedora"]:
+                    pkg_manager = 'yum'
+                else:
+                    pkg_manager = 'apt'  # assume apt
+        elif key in VALID_DOCKER_KEYS:
+            spec_final = deepcopy(DEFAULT_INSTRUCTIONS.get(key, {}))
+            spec_final.update(spec)
+            this_instruction = (key, spec_final)
+            if key == "miniconda":
+                miniconda_env = spec_final["create_env"]
+        else:
+            raise Exception(f"{key} is not a valid key, must be "
+                            f"from the list {VALID_DOCKER_KEYS}")
+        instructions.append(this_instruction)
+
+    # adding post build part
+    post_build = params.get("post_build", {})
```

Comment:
for post build it should do `nfm install` to install the package inside the container and then add the new command to the `entrypoint` - i think this bit should be automated for every niflow.

Reply:
what exactly should be the command to the `entrypoint`

Reply:
i think this should be explicitly added to the spec. the current nfm init adds an automatic command line tool.

Reply:
ok, I've added https://github.com/niflows/niflow-manager/pull/26/commits/64a3c32aecc8a71a15a27bed0e201106c7d60acf